File: 001/pseudoinverse.py
import numpy as np
import timeit

def linear_regression():
    X = np.array([[1, 0], [1, 1], [1, 2], [1, 3]])
    y = np.array([[0], [1], [4], [9]])
    return np.linalg.inv(X.transpose()@X) @ (X.transpose()@y)

trials = 10
print("[Python] Avg Time (s):", timeit.Timer(linear_regression).timeit(number=trials)/trials)

File: 001/pseudoinverse.jl
function linearRegression()
    X = [1 0; 1 1; 1 2; 1 3]
    y = reshape([0; 1; 4; 9], :, 1)
    inv(transpose(X)*X)*(transpose(X)*y)
end

times = UInt64[]
for i in 1:10
    beginning = time_ns()
    linearRegression()
    push!(times, time_ns() - beginning)
end

println("[Julia] Avg Time (s): ", sum(times)/length(times) / 1000000000)
